Index: C:/Users/jfs/Dev/Aegisub/vsfilter/subtitles/RTS.cpp
===================================================================
--- C:/Users/jfs/Dev/Aegisub/vsfilter/subtitles/RTS.cpp	(revision 2284)
+++ C:/Users/jfs/Dev/Aegisub/vsfilter/subtitles/RTS.cpp	(revision 2285)
@@ -703,7 +703,9 @@
 
 			if(w->m_style.borderStyle == 0)
 			{
-				bbox |= w->Draw(spd, clipRect, pAlphaMask, x, y, sw, false, true);
+				// Always draw border part of widened region
+				// Draw fill part of widened region only if there isn't a chance the real fill is transculent
+				bbox |= w->Draw(spd, clipRect, pAlphaMask, x, y, sw, !(w->m_style.alpha[0]||w->m_style.alpha[1]||alpha), true);
 			}
 			else if(w->m_style.borderStyle == 1 && w->m_pOpaqueBox)
 			{
